File: 0301_0400/0330/0330.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = [int(x) for x in f.read().split()]
    x, y = abs(s[0] - s[2]), abs(s[1] - s[3])
    q.write('1' if x == y else '2' if x % 2 == y % 2 else '0')

File: 0301_0400/0330/0330.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int x1, y1, x2, y2;

	fscanf(f, "%d%d%d%d", &x1, &y1, &x2, &y2);
	x1 = abs(x1 - x2), y1 = abs(y1 - y2);
	fprintf(q, "%d", x1 == y1 ? 1 : x1 % 2 == y1 % 2 ? 2 : 0);
	return 0;
}
